modules/rosa-sts-terraform-prerequisites.adoc

```diff
@@ -0,0 +1,113 @@
+ifdef::tf-full[]
++
+You must have Terraform configured for your local system. The Terraform website contains installation options for MacOS, Windows, and Linux.
+endif::tf-full[]
+ifndef::tf-full[]
+* You have created your AWS account-wide IAM roles.
++
+The specific account-wide IAM roles and policies provide the STS permissions required for ROSA support, installation, control plane, and compute functionality. This includes account-wide Operator policies. See the Additional resources for more information on the AWS account roles.
+endif::tf-full[]
+* You must have, at minimum, the following permissions in your AWS IAM role policy that is operating for Terraform.
++
+.Minimum AWS permissions for Terraform
+[%collapsible]
+====
+[source,json]
+----
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "iam:GetPolicyVersion",
+                "iam:DeletePolicyVersion",
+                "iam:CreatePolicyVersion",
+                "iam:UpdateAssumeRolePolicy",
+                "secretsmanager:DescribeSecret",
+                "iam:ListRoleTags",
+                "secretsmanager:PutSecretValue",
+                "secretsmanager:CreateSecret",
+                "iam:TagRole",
+                "secretsmanager:DeleteSecret",
+                "iam:UpdateOpenIDConnectProviderThumbprint",
+                "iam:DeletePolicy",
+                "iam:CreateRole",
+                "iam:AttachRolePolicy",
+                "iam:ListInstanceProfilesForRole",
+                "secretsmanager:GetSecretValue",
+                "iam:DetachRolePolicy",
+                "iam:ListAttachedRolePolicies",
+                "iam:ListPolicyTags",
+                "iam:ListRolePolicies",
+                "iam:DeleteOpenIDConnectProvider",
+                "iam:DeleteInstanceProfile",
+                "iam:GetRole",
+                "iam:GetPolicy",
+                "iam:ListEntitiesForPolicy",
+                "iam:DeleteRole",
+                "iam:TagPolicy",
+                "iam:CreateOpenIDConnectProvider",
+                "iam:CreatePolicy",
+                "secretsmanager:GetResourcePolicy",
+                "iam:ListPolicyVersions",
+                "iam:UpdateRole",
+                "iam:GetOpenIDConnectProvider",
+                "iam:TagOpenIDConnectProvider",
+                "secretsmanager:TagResource"
+            ],
+            "Resource": [
+                "arn:aws:secretsmanager:*:<ACCOUNT_ID>:secret:*",
+                "arn:aws:iam::<ACCOUNT_ID>:instance-profile/*",
+                "arn:aws:iam::<ACCOUNT_ID>:role/*",
+                "arn:aws:iam::<ACCOUNT_ID>:oidc-provider/*",
+                "arn:aws:iam::<ACCOUNT_ID>:policy/*"
+            ]
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": [
+                "s3:PutObject",
```

Comment:
@amalykhi did you change it? 
Please update here your changes

Reply:
Yes, @EricPonvelle could you please update according to this: https://github.com/terraform-redhat/terraform-provider-rhcs/blob/main/docs/index.md?plain=1#L165
All S3 permissions added because of a lot of dependencies in AWS buckets

Reply:
@bardielle and @amalykhi,

I updated the permissions block. Can you check me again to make sure it's correct this time?

@nirarg with regards to the example, this page is solely to be an orientation for Terraform. I have another PR in flight that creates a cluster for the Quick Start Guide, but I think we should leverage the Terraform registry to document all of the examples we may have. Product documentation should be around procedures.